### ISSUE TITLE
Feature request: apply PostgreSQL function on mutation/query

### ISSUE THREAD

**Issue by sanderhahn** [hasura/graphql-engine#4917]
Sometimes you want to store and retrieve sensitive data such as passwords using pgcrypt that is already installed by default. People find this difficult to do using triggers, maybe Hasura could allow configuring a function or snipped to be performed on the data on mutations and queries in the metadata. For instance:

```sql
create table account (
    email text primary key,
    password text
);

insert into account (email, password) values ('you@domain.com', crypt('password', gen_salt('bf')));

select * from account where email = 'you@domain.com' and crypt('password', password) = password;
```

The snippets would be:

```
mutation password -> crypt('password', gen_salt('bf'))
query password -> crypt('password', password) = password
```

Snippets are flexible, but another simpler option would be to be able to mark the field as password field in the metadata and support this way of mutation/query without exposing the underlying sql.

**Comment by mentatxx** [comment]
Yeah, looking forward to run functions via mutations. It is not cool to create actions/lambdas to just call a function.

Volatile functions for queries are not intended to modify data

**Comment by jadiaheno** [comment]
+1 for this. Anything new on this? It will basically help me not to create actions for login/signin/forgot-password

**Comment by Zerebokep** [comment]
Any update on this?

**Comment by rayhantr** [comment]
**Any update?**

![image](https://user-images.githubusercontent.com/49080349/201518675-07065b8a-7ca9-437e-b6ed-2125cbf1294f.png)
